students/elmar_m/lesson04/mailroom.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+
+'''
+file: mailroom.py
+elmar_m / 22e88@mailbox.org
+Lesson04: Mailroom Exercise Part 2
+'''
+
+import time
+import collections
+
+
+main_p = '\n> Main menu. Options: t == thankyou | r == report | q == quit program\n'
+sub_p = '\n>> Sub menu "thankyou". Options: l == list current donors | a == add donation / donor | m == send mail to every donor | x == exit submenu\n'
+
+donors = collections.defaultdict(list)
+ 
+donors['bill'] = [2000, 7.5, 950000]
+donors['steve'] = [5.5, 234000, 928]
+donors['donald'] = [657, 234, 28.57, 90456]
+donors['angie'] = [2, 99, 297765, 47, 28346]
+donors['kim'] = [38982, 66.23, 9856, 0.1]
+
+
+def writefile(name, content):
+    '''
+    Create mail files
+    
+        ARGS:
+    name: filename / name of donor
+    content: mail body text
+    '''
+    ts = time.strftime('%Y%m%d-%H%M%S')
+    fname = '{}.{}.txt'.format(name, ts)
+    f = open(fname, 'w')
+    f.write(content)
+    f.close
```

Comment:
prefer to use `with` when working with file, this ensures that the file is closed